### ISSUE TITLE
Linking errors on Ubuntu 17.04 x86_64

### ISSUE THREAD

**Issue by xavierleroy** [AbsInt/CompCert#189]
As reported by @monniaux: 
```
$ cat foo.c
#include <stdio.h>

int main() {
	printf("toto %d\n", 3);
}

$ /opt/CompCert/3.0.1/bin/ccomp foo.c -o foo
/usr/bin/ld: foo.o: relocation R_X86_64_PC32 against symbol `printf@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
Error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by xavierleroy** [comment]
The root cause is that in Ubuntu 16.10 and 17.04, `gcc` links in PIE mode by default, while the code generated by CompCert is not compatible with PIE.  A quick fix is to add `-no-pie` to the link-time options.  In the longer term, we should look at supporting PIE and PIC in CompCert, at least for x86-64 where it is relatively easy to do so.

**Comment by bschommer** [comment]
I thinke we should avoid adding `-no-pie` since older `gcc` don't know that option, for example the 4.8 version which is still used by many distributions.

**Comment by xavierleroy** [comment]
Well, we need to test in `configure` for availability of `-no-pie`, like we do for MacOS X.  Can we assume that GCC 5.0 and above support `-no-pie` ?  Ubuntu 16.04 has gcc 5.4 and 17.04 has 6.3, so that sounds pretty safe.

**Comment by bschommer** [comment]
That would be okay. According to the gcc manual 5.4 does not have the `-no-pie` option. The option seems to be introduced with gcc 6.0, since that gcc introduced the configure option ` --enable-default-pie`.

**Comment by xavierleroy** [comment]
Often the `-no-` options are not explicitly documented.  gcc 5.4 has `-pie` (documented) and `-no-pie` (undocumented).

**Comment by bschommer** [comment]
I'm not sure, since the older gcc also have the `-pie` option but explicitly no `-no-pie` and I tried it with a gcc 5.2 and it did not recognizes the `-no-pie` option.

**Comment by xavierleroy** [comment]
gcc 5.4 here knows about `-no-pie`.  What a mess...

**Comment by bschommer** [comment]
Could be the case that they have a patched version, I think the easiest way is to have a configure test using this option instead of relying on the version number.

**Comment by bschommer** [comment]
Should be solved by commit 7f05ce9.
